pkg/async/pool/pool.go

```diff
@@ -210,7 +215,7 @@ func (p *Pool) worker(ctx context.Context) {
 	for {
 		select {
 		case u = <-p.queue:
-			//nolint:errcheck
+			//nolint:errcheck // Why: best effort
```

Comment:
This is a pretty big feature gap in the pool! Outside the scope of this PR, but does it really just silently swallow all errors???

Reply:
I think the expectation is that the runner does full error handling here and the pool just tries to execute what's passed to it.